pkg/util/kubernetes/kubelet/podwatcher.go

```diff
@@ -63,27 +69,32 @@ func (w *PodWatcher) computeChanges(podList []*Pod) ([]*Pod, error) {
 			continue
 		}
 
+		var newStaticPod, phaseChange bool
 		podEntity := PodUIDToEntityName(pod.Metadata.UID)
-		newStaticPod := false
-		_, found := w.lastSeen[podEntity]
+		prevSighting, found := w.lastSeen[podEntity]
+
 		// static pods are included specifically because they won't have any container
 		// as they're not updated in the pod list after creation
-		if isPodStatic(pod) == true && found == false {
-			newStaticPod = true
+		if found == false {
+			if isPodStatic(pod) == true {
+				newStaticPod = true
+			} else if prevSighting.phase != pod.Status.Phase {
```

Comment:
If `found == false`, there should be no previous sighting. Was that meant to be when `found == true` instead?